### PR TITLE
factory method gino.ext.starlette

### PR DESCRIPTION
**Pull request by jim-obrien-orig**
Change `gino.ext.starlette` to use factory method like the other extensions, as well as `__init__`. Now init with either `Gino(app, ...)` or `db = Gino(); db.init_app(app)`

### CODE REVIEW

File: tests/test_starlette.py [hunk 1]
```diff
@@ -11,7 +11,6 @@
 from starlette.applications import Starlette
 from starlette.responses import JSONResponse, PlainTextResponse
 from starlette.testclient import TestClient
-
```
Comment (wwwjfy) [inline tests/test_starlette.py]: should keep this line to separate 3rd party imports and local imports

File: tests/test_starlette.py [hunk 2]
```diff
@@ -74,6 +78,8 @@ class User(db.Model):
 
     e = await gino.create_engine(PG_URL)
     try:
+        if factory:
```
Comment (wwwjfy) [inline tests/test_starlette.py]: can we put it up to the previous `if factory` block?

Comment (jim-obrien-orig) [reply tests/test_starlette.py]: I think so - you may know better but I think I would have to move the `create_engine` statement up as well.

Comment (wwwjfy) [reply tests/test_starlette.py]: No. The engine here is just to call `create_all` and not related to app initialization. As you can see, `e` is not passed to `init_app`

Comment (jim-obrien-orig) [reply tests/test_starlette.py]: Oh - wasn't sure if the init_app depended on having an engine. Thanks!

**Comment by wwwjfy** [comment]
Could you revert the format change made in the last commit? It makes the diff less readable and single line longer than 79 characters.

**Comment by jim-obrien-orig** [comment]
Sorry about that. Vim formatted on save. Don't usually use it for python.

**Comment by wwwjfy** [comment]
Thanks!

**Pull request merged**